x/oracle/internal/types/vote_test.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestParseExchangeRateTuples(t *testing.T) {
+	valid := "123.0uluna,123.123ukrw"
+	_, err := ParseExchangeRateTuples(valid)
+	require.NoError(t, err)
+
+	duplicatedDenom := "100.0uluna,123.123ukrw,121233.123ukrw"
+	_, err = ParseExchangeRateTuples(duplicatedDenom)
+	require.Error(t, err)
+
+	invalidCoin := "123.123"
+	_, err = ParseExchangeRateTuples(invalidCoin)
+	require.Error(t, err)
+
+	invalidCoinWithValid := "123.0uluna,123.1"
+	_, err = ParseExchangeRateTuples(invalidCoinWithValid)
+	require.Error(t, err)
```

Comment:
Might want to add a test for empty string, as well as a string containing the denom but with excess characters e.g. "123ukrww", as well as a test containing a strange denom e.g. "123ajdsfjfaddaj"

Reply:
msg validation has no context, so it is hard to test denom is in whitelist or not.
We can do whitelist verification in handler now with voting targets.

Reply:
Makes sense